### PR TITLE
use flex for ContentContainer

### PR DESCRIPTION
**Pull request by Erin007**
Previously, `CourseCard`s had white space between them with the fixed width spacer that was rendered after every left hand card. This change uses `justifyContent: 'space-between'` in `ContentContainer` to achieve that space. The switch was made, in part, to fix a bug that occurred when zooming in on the cards. 

Before: 

scrunched course cards

After: 

zoomed in and spaced course cards

### CODE REVIEW

**Comment by Erin007** [comment]
Everything else in a `ContentContainer` on /home seems to look the same, but another set of eyes would be helpful! Also not 100% sure how this will conflict, if at all, with work on /courses.

File: apps/test/unit/templates/studioHomepages/RecentCoursesTest.js [hunk 1]
```diff
@@ -57,7 +57,6 @@ describe('RecentCourses', () => {
                 link={courses[0].link}
                 isRtl={false}
               />
-              <div>.</div>
```
Comment (breville) [inline apps/test/unit/templates/studioHomepages/RecentCoursesTest.js]: This is another case where I wonder if the test is a bit fragile and is testing (or really just matching) implementation details more than actually testing the outcome of the code...

File: apps/src/templates/ContentContainer.jsx [hunk 2]
```diff
@@ -66,14 +66,14 @@ const styles = {
     float: 'left',
     paddingLeft: 10,
   },
+  children: {
```
Comment (caleybrock) [inline apps/src/templates/ContentContainer.jsx:69]: does this look good with only one item? 
did you take a look at the project widget too?

**Comment by caleybrock** [comment]
It feels like this CSS doesn't belong in the `ContentContainer`. To me, the callers should determine if they want things displayed with space-between or centered, etc, and the `ContentContainer` should be super general and should only manage width and headings of the component. It's not too hard to imagine a component that would want to use these headers but wouldn't want `space-between`. But of course I could be convinced otherwise.

File: apps/src/templates/ContentContainer.jsx (hunk 2 above)
Comment (Erin007) [reply apps/src/templates/ContentContainer.jsx:69]: One item and the Project Widget look the same as before.

**Comment by Erin007** [comment]
@caleybrock I see your point, but I think it makes sense to keep the css here because more often than not (notably every case we're currently using `ContentContainer` for) we will want this space to be part of the container.  In the unlikely, future scenario where we want two or more things without spaces between them to be in a `ContentContainer` we can just wrap those things collectively in a div that handles their spacing, and `ContentContainer` will display them correctly as one thing.

**Comment by Erin007** [comment]
@tanyaparker This is the new PR that fixes the CourseCard spacing.  The closed one #17045 is a different approach that I abandoned in favor of this one.

**Pull request merged**